### PR TITLE
drivers: can: nxp_s32_canxl: calculate and set the TDCO

### LINKED ISSUE #73824: drivers: can: s32: canxl: driver does not set Transceiver Delay Compensation Offset (TDCO)
**Describe the bug**
The NXP S32 CANXL driver does not calculate and set the Transceiver Delay Compensation Offset (TDCO) when CAN FD is enabled. TDCO is needed for correct sampling of own TX line at higher data phase bitrates.

**To Reproduce**
1. Inspect the driver code.
2. Notice TDCO is not set up.

**Expected behavior**
TDCO should be calculated using `CAN_CALC_TDCO()` and written to the respective register and Transceiver Delay Compensation (TDC) should be enabled when using CAN FD.

**Impact**
Likely issues with using higher data phase bitrates.

**Additional context**
Please see the following commits for inspiration on how to calculate and set TDCO:
- https://github.com/zephyrproject-rtos/zephyr/commit/633065ec3b78bc15722b02804ad9d72450ebec71
- https://github.com/zephyrproject-rtos/zephyr/commit/bfad7bc00e734010d059d5ba7eb4cd2187eb1c8d

### PR DESCRIPTION
**Pull request by congnguyenhuu**
Calculate and set the Transceiver Delay Compensation Offset (TDCO).

Fixes #73824

### CODE REVIEW

File: drivers/can/can_nxp_s32_canxl.c [hunk 1]
```diff
@@ -706,6 +708,9 @@ static int can_nxp_s32_set_timing_data(const struct device *dev,
 	/* Set timing for CAN FD instance*/
 	CanXL_SetFDBaudRate(config->base_sic, &can_fd_time_segment);
 
+	Canexcel_Ip_SetTDCOffsetFD(config->instance, true, false,
```
Comment (henrikbrixandersen) [inline drivers/can/can_nxp_s32_canxl.c:711]:
```suggestion
	Canexcel_Ip_SetTDCOffsetFD(config->instance, true, true,
```

File: drivers/can/can_nxp_s32_canxl.c [hunk 2]
```diff
@@ -964,6 +969,11 @@ static int can_nxp_s32_init(const struct device *dev)
 	/* Initialize CAN structure */
 	Canexcel_Ip_Init(config->instance, config->can_cfg, data->can_state);
 
+#ifdef CAN_NXP_S32_FD_MODE
+	Canexcel_Ip_SetTDCOffsetFD(config->instance, true, false,
```
Comment (henrikbrixandersen) [inline drivers/can/can_nxp_s32_canxl.c:973]:
```suggestion
	Canexcel_Ip_SetTDCOffsetFD(config->instance, true, true,
```

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]
You need to enable measurement of the delay as well. The offset calculated by `CAN_CALC_TDCO()` only accounts for the sample point location.

**Comment by henrikbrixandersen** [comment]
Hmm, but based on the code below (from `../modules/hal/nxp/s32/drivers/s32ze/Can_CANEXCEL/include/CanEXCEL_Ip_HwAccess.h`), the `TDCMEnable` seems to do the opposite of what the name implies?

```
static inline void CanXL_SetTDCOffsetFD(CANXL_SIC_Type * base, boolean TDCEnable, boolean TDCMEnable, uint8 Offset)
{
    SchM_Enter_Can_43_CANEXCEL_CAN_EXCLUSIVE_AREA_10();
    base->BTDCC &= ~(CANXL_SIC_BTDCC_FTDCEN_MASK | CANXL_SIC_BTDCC_FTDMDIS_MASK | CANXL_SIC_BTDCC_FTDCOFF_MASK);
    base->BTDCC |= (CANXL_SIC_BTDCC_FTDCEN(TDCEnable ? 1UL : 0UL) | CANXL_SIC_BTDCC_FTDMDIS(TDCMEnable ? 1UL : 0UL) | CANXL_SIC_BTDCC_FTDCOFF(Offset));
    SchM_Exit_Can_43_CANEXCEL_CAN_EXCLUSIVE_AREA_10();
}
```

**Comment by henrikbrixandersen** [comment]
> You need to enable measurement of the delay as well. The offset calculated by `CAN_CALC_TDCO()` only accounts for the sample point location.

If my code analysis above is correct, please disregard my code change suggestions.

**Comment by congnguyenhuu** [comment]
because of some reasons about repo source branch. I closed this PR and I will open later.

**Pull request closed without merging**